src/test/java/hudson/plugins/git/extensions/impl/PruneStaleTagTest.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020 Nikolas Falco
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+package hudson.plugins.git.extensions.impl;
+
+import hudson.EnvVars;
+import hudson.Functions;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitSCM;
+import hudson.util.LogTaskListener;
+import java.io.File;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import nl.jqno.equalsverifier.EqualsVerifier;
+import org.apache.commons.io.FileUtils;
+import org.jenkinsci.plugins.gitclient.GitClient;
+import org.jenkinsci.plugins.gitclient.TestCliGitAPIImpl;
+import org.jenkinsci.plugins.gitclient.TestJGitAPIImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.mockito.Mockito.*;
+
+public class PruneStaleTagTest {
```

Comment:
When I started the test cases the example I saw simply did not extend GitSCMExtensionTest.
Now I checked the GitSCMExtensionTest class but since I do not need neither the Jenkins Rule neither a test project setup does not make sense for me extends it. This test case just work with pure git repo commands.